payjoin-ffi/src/send/mod.rs

```diff
@@ -355,22 +355,14 @@ impl WithReplyKey {
     pub fn process_response(
         &self,
         response: &[u8],
-        post_ctx: Arc<V2PostContext>,
+        post_ctx: &ClientResponse,
     ) -> WithReplyKeyTransition {
-        let mut guard = post_ctx.0.write().expect("Lock should not be poisoned");
-        let post_ctx = guard.take().expect("Value should not be taken");
         WithReplyKeyTransition(Arc::new(RwLock::new(Some(
-            self.clone().0.process_response(response, post_ctx),
+            self.0.clone().process_response(response, post_ctx.into()),
```

Comment:
For reviewer: note this is the same exact pattern as the PollingForProposal process_response pattern.